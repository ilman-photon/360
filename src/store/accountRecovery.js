import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import { Api } from "../pages/api/api";

export const fetchPatientAccount = createAsyncThunk(
  "accountRecovery/fetchPatientAccount",
  async ({ keyword }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/getPatientDetails/${keyword}`;
    return api.getResponse(url, null, "get");
  }
);

export const onActivate = createAsyncThunk(
  "accountRecovery/onActivate",
  async ({ patientId }) => {
    const url = `/api/dummy/account-recovery/onActivate/${patientId}`;
    return fetch(url, { method: "POST" }).then((res) => res.json());
  }
);

export const onUnlock = createAsyncThunk(
  "accountRecovery/onUnlock",
  async ({ patientId }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/unlockAccountByAdmin/${patientId}`;

    try {
      const response = await api.getResponse(url, null, "get");

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

export const onSendPasswordReset = createAsyncThunk(
  "accountRecovery/onSendPasswordReset",
  async ({ patientId, patientData, selectedCommunication }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/sendPasswordResetLinkToUser`;
    const domain = window.location.origin;
    const NEXT_PUBLIC_ONE_TIME_LINK = process.env.NEXT_PUBLIC_ONE_TIME_LINK;
    const postBody = {
      patient: {
        userName: patientData.username,
      },
      preferredComunication: selectedCommunication,
      resetPassword: true,
      link: `${domain}${NEXT_PUBLIC_ONE_TIME_LINK}`,
    };

    try {
      const response = await api.getResponse(url, postBody, "post");

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

export const onShareUsername = createAsyncThunk(
  "accountRecovery/onShareUsername",
  async ({ patientId, selectedCommunication }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/shareUserNameByAdmin`;
    const postBody = {
      patient: {
        id: patientId,
      },
      preferredComunication: selectedCommunication,
    };

    try {
      const response = await api.getResponse(url, postBody, "post");

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

export const onViewSecurityQuestions = createAsyncThunk(
  "accountRecovery/onViewSecurityQuestions",
  async ({ patientId }) => {
    const api = new Api();
    const url = `/ecp/accountRecovery/viewSecurityQuestions/${patientId}`; // d3724cd1-ebae-4f1a-82a5-544ff33b0313

    try {
      const response = await api.getResponse(url, null, "get", false);

      return {
        success: true,
        response,
      };
    } catch (error) {
      console.error({ error });
      return {
        success: false,
        response: error,
      };
    }
  }
);

const getPatientStatus = (data) => {
  if (data.verified != "active") return "inactive";
  switch (data.status) {
    case "locked":
    case "L":
      return "locked";
    case "unlocked":
    default:
      return "unlocked";
  }
};

const buildAccountData = (data) => {
  return data.map((item) => {
    return {
      ...item,
      name: item.patientName || item.name,
      email: item.emailId,
      phone: item.phoneNumber,
      status: getPatientStatus(item),
    };
  });
};

const INITIAL_STATE = {
  patientList: [],
  securityQuestions: [],
  securityQuestionsRaw: {},
  status: null,
  error: null,
};

export const accountRecoveryStore = createSlice({
  name: "index",
  initialState: INITIAL_STATE,
  reducers: {
    setStatus: (state, { payload }) => {
      state.status = payload;
    },
    setAccountDataById: (state, { payload }) => {
      const updatedState = state.patientList.map((item, idx) => {
        const id = item.id || item.patientId;
        const payloadId = payload.id || payload.patientId;
        if (payloadId === id) {
          item = payload;
        }
        return item;
      });

      state.patientList = buildAccountData(updatedState);
    },
  },
  extraReducers: {
    [fetchPatientAccount.pending]: (state) => {
      state.status = "loading";
    },
    [fetchPatientAccount.fulfilled]: (state, { payload }) => {
      state.patientList = buildAccountData(payload);
      state.status = "success";
    },
    [fetchPatientAccount.rejected]: (state, action) => {
      state.error = action;
      state.status = "failed";
    },
    [onViewSecurityQuestions.fulfilled]: (state, { payload }) => {
      const response = payload.response;
      let mapped = [];
      if (response?.ResponseCode === 3000) {
        const securityQuestions = response.SecurityQuestions[0];
        mapped = Object.keys(securityQuestions).map((keys, item) => {
          return {
            question: keys,
            answer: securityQuestions[keys],
          };
        });

        state.securityQuestionsRaw = response.SecurityQuestions;
      }

      state.securityQuestions = mapped;
      state.status = "success";
    },
    [fetchPatientAccount.rejected]: (state, action) => {
      state.error = action;
      state.securityQuestions = [];
    },
  },
});

// Action creators are generated for each case reducer function
export const { setAccountDataById, setStatus } = accountRecoveryStore.actions;

export default accountRecoveryStore.reducer;
