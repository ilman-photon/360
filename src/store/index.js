import { createSlice } from "@reduxjs/toolkit";

const INITIAL_STATE = {
  loading: true,
  counter: 0,
};

export const indexStore = createSlice({
  name: "index",
  initialState: INITIAL_STATE,
  reducers: {
    setLoading: (state, action) => {
      state.loading = action.payload;
    },
    increment: (state, action) => {
      state.counter++;
    },
  },
});

// Action creators are generated for each case reducer function
export const { setLoading, increment } = indexStore.actions;

export default indexStore.reducer;
