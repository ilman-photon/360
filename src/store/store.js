import { configureStore } from "@reduxjs/toolkit";
import index from "../store/index";
import user from "../store/user";

export const store = configureStore({
  reducer: {
    user,
    index,
  },
});

export default store;
