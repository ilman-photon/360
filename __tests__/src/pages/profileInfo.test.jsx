import { render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import ProfileInformationPage from "../../../src/pages/patient/account/profile-info";

import { Provider } from "react-redux";
import store from "../../../src/store/store";

describe("ProfileInformationPage Components", () => {
  let container;
  beforeEach(async() => {
    container = render(<Provider store={store}><ProfileInformationPage 
    /></Provider>);
    await waitFor(() => container.getByText("Profile"));
    await waitFor(() => container.getByText("Contact"));
    await waitFor(() => container.getByRole("button", { name: /Edit/i }));
  });

  it("ProfileInformationPage render", () => {
    expect(container).toMatchSnapshot();
  });


});
