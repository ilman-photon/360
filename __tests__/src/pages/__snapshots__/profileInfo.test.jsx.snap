// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInformationPage Components ProfileInformationPage render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="MuiTabs-root css-4km1ta-MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            style="overflow: hidden; margin-bottom: 0px;"
          >
            <div
              class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-controls="full-width-tabpanel-0"
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-fullWidth Mui-selected css-1vw5i9y-MuiButtonBase-root-MuiTab-root"
                id="full-width-tab-0"
                role="tab"
                tabindex="0"
                type="button"
              >
                Profile
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-controls="full-width-tabpanel-1"
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-fullWidth css-1vw5i9y-MuiButtonBase-root-MuiTab-root"
                id="full-width-tab-1"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Contact
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root card css-1v5z4dq-MuiPaper-root-MuiCard-root"
            >
              <div
                class="MuiCardContent-root css-vp7ejv-MuiCardContent-root"
              >
                <div
                  class="css-1lwbda4-MuiStack-root"
                  style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <div
                    class="css-1p5q5e5-MuiStack-root"
                  >
                    <div
                      class="labelContainer"
                    >
                      <div
                        class="label"
                      >
                        Photo
                      </div>
                    </div>
                    <div
                      class="css-1okj3ks-MuiStack-root"
                    >
                      <div
                        style="margin-top: 10px;"
                      >
                        <div
                          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2wkcz1-MuiAvatar-root"
                        >
                          EU
                        </div>
                      </div>
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                  />
                  <div
                    class="css-1p5q5e5-MuiStack-root"
                  >
                    <div
                      class="labelContainer"
                    >
                      <div
                        class="label"
                      >
                        Name
                      </div>
                      <svg
                        aria-hidden="true"
                        aria-label="If you wish to change this information, please contact Customer Support"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yagxpu-MuiSvgIcon-root"
                        data-mui-internal-clone-element="true"
                        data-testid="ErrorOutlineOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                    </div>
                    <div
                      class="css-1okj3ks-MuiStack-root"
                    >
                      <div
                        style="margin-top: 10px;"
                      />
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                  />
                  <div
                    class="css-1p5q5e5-MuiStack-root"
                  >
                    <div
                      class="labelContainer"
                    >
                      <div
                        class="label"
                      >
                        Preferred Name
                      </div>
                    </div>
                    <div
                      class="css-1okj3ks-MuiStack-root"
                    >
                      <div
                        style="margin-top: 10px;"
                      >
                        ---
                      </div>
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                  />
                  <div
                    class="css-1p5q5e5-MuiStack-root"
                  >
                    <div
                      class="labelContainer"
                    >
                      <div
                        class="label"
                      >
                        Title
                      </div>
                    </div>
                    <div
                      class="css-1okj3ks-MuiStack-root"
                    >
                      <div
                        style="margin-top: 10px;"
                      />
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                  />
                  <div
                    class="css-1p5q5e5-MuiStack-root"
                  >
                    <div
                      class="labelContainer"
                    >
                      <div
                        class="label"
                      >
                        Date of Birth
                      </div>
                      <svg
                        aria-hidden="true"
                        aria-label="If you wish to change this information, please contact Customer Support"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yagxpu-MuiSvgIcon-root"
                        data-mui-internal-clone-element="true"
                        data-testid="ErrorOutlineOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                    </div>
                    <div
                      class="css-1okj3ks-MuiStack-root"
                    >
                      <div
                        style="margin-top: 10px;"
                      >
                        Invalid Date
                      </div>
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                  />
                  <div
                    class="css-1p5q5e5-MuiStack-root"
                  >
                    <div
                      class="labelContainer"
                    >
                      <div
                        class="label"
                      >
                        Age
                      </div>
                      <svg
                        aria-hidden="true"
                        aria-label="If you wish to change this information, please contact Customer Support"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yagxpu-MuiSvgIcon-root"
                        data-mui-internal-clone-element="true"
                        data-testid="ErrorOutlineOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                    </div>
                    <div
                      class="css-1okj3ks-MuiStack-root"
                    >
                      <div
                        style="margin-top: 10px;"
                      />
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                  />
                  <div
                    class="css-1p5q5e5-MuiStack-root"
                  >
                    <div
                      class="labelContainer"
                    >
                      <div
                        class="label"
                      >
                        Gender
                      </div>
                    </div>
                    <div
                      class="css-1okj3ks-MuiStack-root"
                    >
                      <div
                        style="margin-top: 10px;"
                      />
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                  />
                  <div
                    class="css-1p5q5e5-MuiStack-root"
                  >
                    <div
                      class="labelContainer"
                    >
                      <div
                        class="label"
                      >
                        SSN
                      </div>
                      <svg
                        aria-hidden="true"
                        aria-label="If you wish to change this information, please contact Customer Support"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yagxpu-MuiSvgIcon-root"
                        data-mui-internal-clone-element="true"
                        data-testid="ErrorOutlineOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                    </div>
                    <div
                      class="css-1okj3ks-MuiStack-root"
                    >
                      <div
                        style="margin-top: 10px;"
                      />
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                  />
                  <div>
                    <h3
                      class="MuiTypography-root MuiTypography-h3 css-1mie9p3-MuiTypography-root"
                    >
                      State Issued ID
                    </h3>
                    <div
                      class="MuiTypography-root MuiTypography-bodyRegular css-15atyh3-MuiTypography-root"
                    >
                      Please upload a photo of government-issued ID, such as Driver’s License or State-issued ID.
                    </div>
                    <div
                      class="css-14w0e8k-MuiStack-root"
                    >
                      <div
                        class="css-1p5q5e5-MuiStack-root"
                      >
                        <div
                          class="labelContainer"
                        >
                          <div
                            class="label"
                          >
                            Front Card
                          </div>
                        </div>
                        <div
                          class="css-1okj3ks-MuiStack-root"
                        >
                          <div
                            style="margin-top: 10px;"
                          >
                            <div
                              class="issuedCardContainer"
                            >
                              <img
                                alt="front-id"
                                height="175"
                                width="267"
                              />
                            </div>
                          </div>
                          <span
                            class="MuiTypography-root MuiTypography-bodySmallRegular css-1r97m9f-MuiTypography-root"
                          >
                            JPG or PNG file formats only. (File size limit is 4 MB)
                          </span>
                        </div>
                      </div>
                      <div
                        class="css-1p5q5e5-MuiStack-root"
                      >
                        <div
                          class="labelContainer"
                        >
                          <div
                            class="label"
                          >
                            Back Card
                          </div>
                        </div>
                        <div
                          class="css-1okj3ks-MuiStack-root"
                        >
                          <div
                            style="margin-top: 10px;"
                          >
                            <div
                              class="issuedCardContainer"
                            >
                              <img
                                alt="back-id"
                                height="175"
                                width="267"
                              />
                            </div>
                          </div>
                          <span
                            class="MuiTypography-root MuiTypography-bodySmallRegular css-1r97m9f-MuiTypography-root"
                          >
                            JPG or PNG file formats only. (File size limit is 4 MB)
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="position: fixed; bottom: 16px; right: 16px;"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root customButton customButtonSmall  css-2ak3pm-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Edit
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="MuiTabs-root css-4km1ta-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-controls="full-width-tabpanel-0"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-fullWidth Mui-selected css-1vw5i9y-MuiButtonBase-root-MuiTab-root"
              id="full-width-tab-0"
              role="tab"
              tabindex="0"
              type="button"
            >
              Profile
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls="full-width-tabpanel-1"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-fullWidth css-1vw5i9y-MuiButtonBase-root-MuiTab-root"
              id="full-width-tab-1"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Contact
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root card css-1v5z4dq-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-vp7ejv-MuiCardContent-root"
            >
              <div
                class="css-1lwbda4-MuiStack-root"
                style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
              >
                <div
                  class="css-1p5q5e5-MuiStack-root"
                >
                  <div
                    class="labelContainer"
                  >
                    <div
                      class="label"
                    >
                      Photo
                    </div>
                  </div>
                  <div
                    class="css-1okj3ks-MuiStack-root"
                  >
                    <div
                      style="margin-top: 10px;"
                    >
                      <div
                        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2wkcz1-MuiAvatar-root"
                      >
                        EU
                      </div>
                    </div>
                  </div>
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <div
                  class="css-1p5q5e5-MuiStack-root"
                >
                  <div
                    class="labelContainer"
                  >
                    <div
                      class="label"
                    >
                      Name
                    </div>
                    <svg
                      aria-hidden="true"
                      aria-label="If you wish to change this information, please contact Customer Support"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yagxpu-MuiSvgIcon-root"
                      data-mui-internal-clone-element="true"
                      data-testid="ErrorOutlineOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                  </div>
                  <div
                    class="css-1okj3ks-MuiStack-root"
                  >
                    <div
                      style="margin-top: 10px;"
                    />
                  </div>
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <div
                  class="css-1p5q5e5-MuiStack-root"
                >
                  <div
                    class="labelContainer"
                  >
                    <div
                      class="label"
                    >
                      Preferred Name
                    </div>
                  </div>
                  <div
                    class="css-1okj3ks-MuiStack-root"
                  >
                    <div
                      style="margin-top: 10px;"
                    >
                      ---
                    </div>
                  </div>
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <div
                  class="css-1p5q5e5-MuiStack-root"
                >
                  <div
                    class="labelContainer"
                  >
                    <div
                      class="label"
                    >
                      Title
                    </div>
                  </div>
                  <div
                    class="css-1okj3ks-MuiStack-root"
                  >
                    <div
                      style="margin-top: 10px;"
                    />
                  </div>
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <div
                  class="css-1p5q5e5-MuiStack-root"
                >
                  <div
                    class="labelContainer"
                  >
                    <div
                      class="label"
                    >
                      Date of Birth
                    </div>
                    <svg
                      aria-hidden="true"
                      aria-label="If you wish to change this information, please contact Customer Support"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yagxpu-MuiSvgIcon-root"
                      data-mui-internal-clone-element="true"
                      data-testid="ErrorOutlineOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                  </div>
                  <div
                    class="css-1okj3ks-MuiStack-root"
                  >
                    <div
                      style="margin-top: 10px;"
                    >
                      Invalid Date
                    </div>
                  </div>
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <div
                  class="css-1p5q5e5-MuiStack-root"
                >
                  <div
                    class="labelContainer"
                  >
                    <div
                      class="label"
                    >
                      Age
                    </div>
                    <svg
                      aria-hidden="true"
                      aria-label="If you wish to change this information, please contact Customer Support"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yagxpu-MuiSvgIcon-root"
                      data-mui-internal-clone-element="true"
                      data-testid="ErrorOutlineOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                  </div>
                  <div
                    class="css-1okj3ks-MuiStack-root"
                  >
                    <div
                      style="margin-top: 10px;"
                    />
                  </div>
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <div
                  class="css-1p5q5e5-MuiStack-root"
                >
                  <div
                    class="labelContainer"
                  >
                    <div
                      class="label"
                    >
                      Gender
                    </div>
                  </div>
                  <div
                    class="css-1okj3ks-MuiStack-root"
                  >
                    <div
                      style="margin-top: 10px;"
                    />
                  </div>
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <div
                  class="css-1p5q5e5-MuiStack-root"
                >
                  <div
                    class="labelContainer"
                  >
                    <div
                      class="label"
                    >
                      SSN
                    </div>
                    <svg
                      aria-hidden="true"
                      aria-label="If you wish to change this information, please contact Customer Support"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yagxpu-MuiSvgIcon-root"
                      data-mui-internal-clone-element="true"
                      data-testid="ErrorOutlineOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                  </div>
                  <div
                    class="css-1okj3ks-MuiStack-root"
                  >
                    <div
                      style="margin-top: 10px;"
                    />
                  </div>
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <div>
                  <h3
                    class="MuiTypography-root MuiTypography-h3 css-1mie9p3-MuiTypography-root"
                  >
                    State Issued ID
                  </h3>
                  <div
                    class="MuiTypography-root MuiTypography-bodyRegular css-15atyh3-MuiTypography-root"
                  >
                    Please upload a photo of government-issued ID, such as Driver’s License or State-issued ID.
                  </div>
                  <div
                    class="css-14w0e8k-MuiStack-root"
                  >
                    <div
                      class="css-1p5q5e5-MuiStack-root"
                    >
                      <div
                        class="labelContainer"
                      >
                        <div
                          class="label"
                        >
                          Front Card
                        </div>
                      </div>
                      <div
                        class="css-1okj3ks-MuiStack-root"
                      >
                        <div
                          style="margin-top: 10px;"
                        >
                          <div
                            class="issuedCardContainer"
                          >
                            <img
                              alt="front-id"
                              height="175"
                              width="267"
                            />
                          </div>
                        </div>
                        <span
                          class="MuiTypography-root MuiTypography-bodySmallRegular css-1r97m9f-MuiTypography-root"
                        >
                          JPG or PNG file formats only. (File size limit is 4 MB)
                        </span>
                      </div>
                    </div>
                    <div
                      class="css-1p5q5e5-MuiStack-root"
                    >
                      <div
                        class="labelContainer"
                      >
                        <div
                          class="label"
                        >
                          Back Card
                        </div>
                      </div>
                      <div
                        class="css-1okj3ks-MuiStack-root"
                      >
                        <div
                          style="margin-top: 10px;"
                        >
                          <div
                            class="issuedCardContainer"
                          >
                            <img
                              alt="back-id"
                              height="175"
                              width="267"
                            />
                          </div>
                        </div>
                        <span
                          class="MuiTypography-root MuiTypography-bodySmallRegular css-1r97m9f-MuiTypography-root"
                        >
                          JPG or PNG file formats only. (File size limit is 4 MB)
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="position: fixed; bottom: 16px; right: 16px;"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root customButton customButtonSmall  css-2ak3pm-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Edit
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
